src/core/meshcop/joiner.cpp

```diff
@@ -67,8 +67,6 @@ Joiner::Joiner(ThreadNetif &aNetif):
     mState(kStateIdle),
     mCallback(NULL),
     mContext(NULL),
```

Comment:
I think it is fine to initialize them to zero.  I prefer to have all member variables initialized in the constructor regardless of how they are used.  It helps prevent future bugs and makes debugging easier by having the variables in a known state.